app/src/background.js

```diff
@@ -1,75 +1,94 @@
-function getVideoFileName({lesson}, downloadHD) {
-  // ES6 allows you to do this.
-  // Old: const updatedAt = lesson.lesson.updatedAt;
-  // Old: const age, name = person.age, person.name
-  // New: const {age, name} = person;
-  const {updatedAt} = lesson.video.media;
-  const quality = (downloadHD) ? "_HD" : "_SD";
-  return updatedAt.slice(0, updatedAt.indexOf("T")) + quality + ".mp4";
+function getVideoFileName({ lesson }) {
+  const dateObj = new Date(lesson.startTimeUTC);
+  const month = dateObj.getUTCMonth() + 1; //months from 1-12
+  const day = dateObj.getUTCDate();
+  const year = dateObj.getUTCFullYear();
+  
+  const startTimeDateObj = new Date(lesson.lesson.timing.start);
+  const startTime = startTimeDateObj.toLocaleTimeString(navigator.language, {
+    hour: '2-digit',
+    minute:'2-digit'
+  });
+
+
+  const endTimeDateObj = new Date(lesson.lesson.timing.end);
+  const endTime = endTimeDateObj.toLocaleTimeString(navigator.language, {
+    hour: '2-digit',
+    minute:'2-digit'
+  });
+
+  const name = year + "-" + month + "-" + day + ' ' + startTime + '-' + endTime;
+
+  return name.replaceAll(':', '\ua789');
 }
 
-// Returns only unit code.
-function getUnitCode({lesson}) {
-  const lectureName = lesson.lesson.name;
-  var unitCodeTrailing = lectureName.slice(0, lectureName.indexOf("/"));
-  var splitChar = "_";
-  if (unitCodeTrailing.indexOf("_") == -1) {
-    // If there aren't underscores, split by space instead (for e.g. UNSW).
-    if (unitCodeTrailing.indexOf(" ") != -1) {
-      splitChar = " "; 
-    }
-  }
-  try {
-    return unitCodeTrailing.split(splitChar)[0];
-  } catch (err) {
-    // Some Universities may have weird formats.
-    return unitCodeTrailing;
+async function getDownloadLink({ lesson }, downloadHD) {
+  const regex = /(?:\(\")(?:.*)(?:\"\))/;
+  const lessonHTMLPageRequest = new Request(`https://echo360.org.uk/lesson/${lesson.lesson.id}/classroom`, { method: 'GET', credentials: 'include' });
```

Comment:
Originally we had a regex here which allows more than one echo360 domain to be support (i.e. echo360.org, echo360.org.au, etc.)

Reply:
done